File: healthchecks/payment-service/src/main/java/com/varunu28/paymentservice/model/Payment.java
package com.varunu28.paymentservice.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.util.Date;
import java.util.UUID;

@Entity
@Table(name = "payments")
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private UUID id;

    private String idempotencyKey;
    private double amount;
    private UUID customerId;
    private Date createdAt;
    private Date updatedAt;

    public Payment() {
    }

    public Payment(String idempotencyKey, double amount, UUID customerId) {
        this.idempotencyKey = idempotencyKey;
        this.amount = amount;
        this.customerId = customerId;
        this.createdAt = new Date();
        this.updatedAt = new Date();
    }

    public UUID getId() {
        return id;
    }

    public String getIdempotencyKey() {
        return idempotencyKey;
    }

    public double getAmount() {
        return amount;
    }

    public UUID getCustomerId() {
        return customerId;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }
}

File: saga/payment-service/src/main/kotlin/com/saga/paymentservice/Payment.kt
package com.saga.paymentservice

import jakarta.persistence.*
import java.math.BigInteger
import java.time.LocalDateTime

@Entity
@Table(name = "payments")
data class Payment(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: BigInteger? = null,
    @Column("customer_name") val customerName: String,
    @Column("order_id") var orderId: BigInteger,
    var amount: Double,
    var status: String,
    @Column("idempotency_key") val idempotencyKey: String,
    @Column("created_at") val createdAt: LocalDateTime,
    @Column("updated_at") var updatedAt: LocalDateTime
)